activerecord/lib/active_record/relation.rb

```diff
@@ -280,20 +280,20 @@ def empty?
     end
 
     # Returns true if there are no records.
-    def none?
-      return super if block_given?
+    def none?(*args)
+      return super if args.present? || block_given?
```

Comment:
```suggestion
      return super if !args.empty? || block_given?
```

Might as well not use AS here.

Reply:
Hmm, imo reads worse, and AR already depends on AS.

Reply:
> AR already depends on AS.

Not on all of it. It only require `active_support`, core ext are required on a per use basis.

Reply:
IMO does not make a difference. AS is here to make it more pleasant to write Ruby, so we should feel free to use it in our internal use as well. and `args.present?` read much nicer to me than `!args.empty?`.